app/streaming/video/ffmpeg-renderers/eglvid.cpp

```diff
@@ -0,0 +1,480 @@
+// vim: noai:ts=4:sw=4:softtabstop=4:expandtab
+#define GL_GLEXT_PROTOTYPES
+
+#include "eglvid.h"
+
+#include "streaming/session.h"
+#include "path.h"
+
+#include <QDir>
+
+#include <Limelight.h>
+#include <unistd.h>
+
+#include <SDL_egl.h>
+#include <SDL_opengl.h>
+#include <SDL_syswm.h>
+#include <SDL_opengles2.h>
+#include <SDL_opengles2_gl2ext.h>
+
+/* TODO:
+ *  - handle more pixel formats
+ *  - code refacto/cleanup
+ *  - handle software decoding
+ *  - handle window resize
+ */
+
+/* DOC/misc:
+ *  - https://kernel-recipes.org/en/2016/talks/video-and-colorspaces/
+ *  - http://www.brucelindbloom.com/
+ *  - https://learnopengl.com/Getting-started/Shaders
+ *  - https://github.com/stunpix/yuvit
+ *  - https://en.wikipedia.org/wiki/YCbCr#ITU-R_BT.601_conversion
+ *  - https://www.renesas.com/eu/en/www/doc/application-note/an9717.pdf
+ *  - https://www.xilinx.com/support/documentation/application_notes/xapp283.pdf
+ *  - https://www.itu.int/dms_pubrec/itu-r/rec/bt/R-REC-BT.709-6-201506-I!!PDF-E.pdf
+ *  - https://www.khronos.org/registry/OpenGL/extensions/OES/OES_EGL_image_external.txt
+ *  - https://gist.github.com/rexguo/6696123
+ *  - https://wiki.libsdl.org/CategoryVideo
+ */
+
+static QStringList egl_get_extensions(EGLDisplay dpy) {
+    const auto egl_extensions_str = eglQueryString(dpy, EGL_EXTENSIONS);
+    if (!egl_extensions_str) {
+        SDL_LogWarn(SDL_LOG_CATEGORY_APPLICATION, "Unable to get EGL extensions");
+        return QStringList();
+    }
+    return QString(egl_extensions_str).split(" ");
+}
+
+EGLExtensions::EGLExtensions(EGLDisplay dpy) :
+    m_extensions(egl_get_extensions(dpy))
+{}
+
+bool EGLExtensions::is_supported(const QString &extension) const {
+    return m_extensions.contains(extension);
+}
+
+EGLRenderer::EGLRenderer(IFFmpegRenderer *frontend_renderer)
+    :
+        m_SwPixelFormat(AV_PIX_FMT_NONE),
+        m_egl_display(nullptr),
+        m_textures{0},
+        m_shader_program(0),
+        m_context(0),
+        m_window(nullptr),
+        m_frontend(frontend_renderer),
+        m_vao(0),
+        m_colorspace(AVCOL_SPC_NB),
+        m_color_full(false),
+        EGLImageTargetTexture2DOES(nullptr)
+{
+    SDL_assert(frontend_renderer);
+    SDL_assert(frontend_renderer->canExportEGL());
+}
+
+EGLRenderer::~EGLRenderer()
+{
+    deinitialize();
+}
+
+bool EGLRenderer::prepareDecoderContext(AVCodecContext*, AVDictionary**)
+{
+    /* Nothing to do */
+
+    SDL_LogInfo(SDL_LOG_CATEGORY_APPLICATION,
+                "Using EGL renderer!!");
+
+    return true;
+}
+
+void EGLRenderer::notifyOverlayUpdated(Overlay::OverlayType)
+{
+    // TODO: FIXME
```

Comment:
This is a minor regression from the current functionality of VAAPI on Wayland today, because the SDL renderer does have overlay support.

Do you think this is something you'd be able to finish up soon in a follow-up PR?

Reply:
To be honest I was kinda hoping someone else would do it :)
I don't really like OpenGL and displaying text seems a **lot** of boiler plate code.

My first approach of this feature (the wole PR) was to use the `SDLRenderer` and use `SDL_Texture` but the `YCbCr` planes cannot be directly bound.

Maybe we could hack around the `SDL_Renderer` to draw the text while doing raw GL before/after. If this works, it would take advantage of undocumented SDL behaviors and could break later, what do you think about it ?

PS: If needed I could do it using raw OpenGL, or better, if you agree I could use another new third party library to handle this (like [imgui](https://github.com/ocornut/imgui) for example)